### ISSUE TITLE
Issue with reading compressed files

### ISSUE THREAD

**Issue by SunderB** [godotengine/godot#64058]
### Godot version

3.5.stable (via Flatpak)

### System information

Debian 11 (Bullseye)

### Issue description

In Godot 3.5, reading compressed files using `open_compressed()` and `get_as_text()` seems to not work. It was working in 3.4. `get_as_text`returns a blank string and gives the following error:
```
E 0:00:01.378   get_as_utf8_string: Condition "r != len" is true. Returned: String()
  <C++ Source>  core/os/file_access.cpp:388 @ get_as_utf8_string()
  <Stack Trace> test.gd:31 @ read_file()
                test.gd:13 @ _ready()
```

### Steps to reproduce

* Open the compressed file test project (provided below)
* Run the main scene
* The files are written in the user folder, but all the tests fail when reading them back, for all compression modes.

### Minimal reproduction project
```gd
extends Spatial

const path_template = "user://test-%s.txt"
const data = "Supercalifragilisticexpialidocius"
const modes = ["FASTLZ","DEFLATE","ZSTD","GZIP"]

func _ready():
    print("User data folder: %s" % OS.get_user_data_dir())
    for i in range(0,4):
        var path = path_template % modes[i]
        write_file(path, i)
        OS.delay_msec(1000) # Give some time for the file to finish being written - may not be needed
        var s = read_file(path, i)
        if (s == data):
            print("Mode %s: ✔" % modes[i])
        else:
            print("Mode %s: X" % modes[i])
            print("Expected '%s', got '%s'" % [data, s])
    
func write_file(path, mode: int):
    var f = File.new()
    if (f.open_compressed(path, File.WRITE, mode) == OK):
        f.store_string(data)
        f.close()
    else:
        print("Error when opening file")
        
func read_file(path, mode: int) -> String:
    var f = File.new()
    if (f.open_compressed(path, File.READ, mode) == OK):
        var s = f.get_as_text()
        f.close()
        return s
    else:
        print("Error when opening file")
        return ""
```
[compressed-file-test.zip](https://github.com/godotengine/godot/files/9277025/compressed-file-test.zip)

**Comment by bruvzg** [comment]
The relevant part of `get_as_utf8_string` was not changed on `3.5`, so it's probably have the same issue in `3.4`, the thing that changed is `get_as_text` which was not using `get_as_utf8_string`, but these changes probably irrelevant.

Most likely, it's something is wrong with compressed file `get_buffer` or `get_len`.

**Comment by bruvzg** [comment]
It's an issue with `get_buffer` returned size, which is fixed in `master` by https://github.com/godotengine/godot/pull/53434.

**Comment by SunderB** [comment]
After cherry-picking e8f7b79b onto the 3.x branch, I can confirm it works again :) Thank you!

**Comment by m4gr3d** [comment]
> After cherry-picking [e8f7b79](https://github.com/godotengine/godot/commit/e8f7b79bb00870a6c6b64f81f9a6d7ecea50d823) onto the 3.x branch, I can confirm it works again :) Thank you!

Looks like the fix was [cherry-picked for 3.5.1 and 3.6](https://github.com/godotengine/godot/pull/53434#issuecomment-1207994116), so the issue should be resolved soon.
